File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        Deque<TreeNode> stack = new ArrayDeque<>();
        List<Integer> ans = new ArrayList<>();
        TreeNode curr = root;

        while(!stack.isEmpty() || curr != null){
            while(curr != null){
                stack.add(curr);
                curr = curr.left;
            }
            curr = stack.removeLast();
            ans.add(curr.val);
            curr = curr.right;
        }

        return ans;

        
    }

    public List<Integer> inorderTraversal(TreeNode root, List<Integer> list){
        if(root == null){
            return list;
        }

        inorderTraversal(root.left, list);
        list.add(root.val);
        inorderTraversal(root.right, list);

        return list;
    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func maxDepth(root *TreeNode) int {
    if root == nil {
		return 0
	}

	max := maxInt(maxDepth(root.Left), maxDepth(root.Right))

	return max +1


}

func maxInt(a, b int) int {
	if a > b{
		return a
	}
	return b
}
